pkg/image/validation.go

```diff
@@ -0,0 +1,153 @@
+package image
+
+import (
+	"fmt"
+	"strings"
+)
+
+func validate(definition *Definition) error {
+	err := validateAPIVersion(definition)
+	if err != nil {
+		return fmt.Errorf("error validating apiVersion: %w", err)
+	}
+	err = validateImage(definition)
+	if err != nil {
+		return fmt.Errorf("error validating image: %w", err)
+	}
+	err = validateOperatingSystem(definition)
+	if err != nil {
+		return fmt.Errorf("error validating operating system: %w", err)
+	}
+
+	return nil
+}
+
+func validateAPIVersion(definition *Definition) error {
+	// Open to discussion regarding how to validate the else if
+	if definition.APIVersion == (Definition{}.APIVersion) {
+		return fmt.Errorf("apiVersion not defined")
+	} else if len(definition.APIVersion) < 2 {
+		return fmt.Errorf("invalid apiVersion")
+	}
+	return nil
+}
+
+func validateImage(definition *Definition) error {
+	if definition.Image == (Image{}) {
+		return fmt.Errorf("image not defined")
+	}
+	if definition.Image.ImageType == (Definition{}.Image.ImageType) {
```

Comment:
We don't need to create zero-valued structs just for the comparison purposes. The zero value of the most inner field (in this case `string`) is just enough to compare. Applies to all other checks below as well.

```suggestion
	if definition.Image.ImageType ==  "") {
```